ibis/backends/bigquery/client.py

```diff
@@ -0,0 +1,277 @@
+"""BigQuery ibis client implementation."""
+
+from __future__ import annotations
+
+import functools
+
+import google.cloud.bigquery as bq
+import pandas as pd
+
+import ibis.common.exceptions as com
+import ibis.expr.datatypes as dt
+import ibis.expr.operations as ops
+import ibis.expr.schema as sch
+from ibis.backends.base import Database
+from ibis.backends.bigquery.datatypes import ibis_type_to_bigquery_type
+
+NATIVE_PARTITION_COL = "_PARTITIONTIME"
+
+
+_DTYPE_TO_IBIS_TYPE = {
+    "INT64": dt.int64,
+    "FLOAT64": dt.double,
+    "BOOL": dt.boolean,
+    "STRING": dt.string,
+    "DATE": dt.date,
+    # FIXME: enforce no tz info
+    "DATETIME": dt.timestamp,
+    "TIME": dt.time,
+    "TIMESTAMP": dt.timestamp,
+    "BYTES": dt.binary,
+    "NUMERIC": dt.Decimal(38, 9),
+}
+
+
+_LEGACY_TO_STANDARD = {
+    "INTEGER": "INT64",
+    "FLOAT": "FLOAT64",
+    "BOOLEAN": "BOOL",
+}
+
+
+@dt.dtype.register(bq.schema.SchemaField)
```

Comment:
Please create a follow-up issue to factor the datatype conversions out to `ibis.backends.bigquery.datatypes`

Reply:
Would you mind if I did that in a follow-up?

Reply:
Not at all, just have an issue to track this

Reply:
Done: https://github.com/ibis-project/ibis/issues/4881